src/playback-watcher.js

```diff
@@ -512,18 +518,36 @@ export default class PlaybackWatcher {
     return false;
   }
 
-  videoUnderflow_(nextRange, buffered, currentTime) {
-    if (nextRange.length === 0) {
+  videoUnderflow_({videoBuffered, audioBuffered, currentTime}) {
+    if (!videoBuffered) {
+      return;
+    }
+    let gap;
+
+    if (videoBuffered.length && audioBuffered.length) {
+      const lastVideoRange = Ranges.findRange(videoBuffered, currentTime - 3);
+      const videoRange = Ranges.findRange(videoBuffered, currentTime);
+      const audioRange = Ranges.findRange(audioBuffered, currentTime);
+
+      if (audioRange.length && !videoRange.length && lastVideoRange.length) {
+        gap = {start: lastVideoRange.end(0), end: audioRange.end(0)};
+      }
+
+    } else {
+      const nextRange = Ranges.findNextRange(audioBuffered, currentTime);
+
       // Even if there is no available next range, there is still a possibility we are
       // stuck in a gap due to video underflow.
-      const gap = this.gapFromVideoUnderflow_(buffered, currentTime);
```

Comment:
and this generic buffered gap check isn't necessary because we still check for this gap above which may not be a video underflow? Or is this equivalent to checking against videoBuffered?

Reply:
The generic `buffered` only worked when the it was referring to muxed content because we will go off of the "video buffer", which is really the muxed content, in that case. Now that we have a case that checks for demuxed content we can just use videoBuffered for the original muxed check and both buffers for the demuxed check.